syncd/scripts/syncd_init_common.sh

```diff
@@ -24,12 +24,20 @@ fi
 # Use temporary view between init and apply
 CMD_ARGS+=" -u"
 
-case "$(cat /proc/cmdline)" in
-  *fast-reboot*)
+BOOT_TYPE="$(cat /proc/cmdline | grep -o 'SONIC_BOOT_TYPE=\S*' | cut -d'=' -f2)"
+
+case "$BOOT_TYPE" in
+  fast-reboot)
```

Comment:
since you are breaking backward compatibility, I suggest to you to change to fast and remove -reboot.